src/broadcast/broadcast.rs

```diff
@@ -43,7 +44,8 @@ pub struct Broadcast<N> {
     /// The hashes and proofs we have received via `Echo` and `EchoHash` messages, by sender ID.
     echos: BTreeMap<N, (Digest, Option<Proof<Vec<u8>>>)>,
     /// The hashes we have received via `CanDecode` messages, by sender ID.
-    can_decodes: BTreeMap<N, Digest>,
+    /// A node can receive conflicting `CanDecode`s from the same node.
+    can_decodes: BTreeSet<(N, Digest)>,
```

Comment:
It might make sense to make this a `BTreeMap<Digest, BTreeSet<N>>` instead: We probably always need to access it by hash? That way we don't need to create those tuples, and clone the root hash inside the loop.